### PR TITLE
Lint accept range

### PR DESCRIPTION
**Pull request by francescayeye**
i messed with #527

### CODE REVIEW

**Comment by aspacca** [comment]
@ochaton sorry for the mess :(

File: server/handlers.go [hunk 1]
```diff
@@ -1029,18 +1039,30 @@ func (s *Server) getHandler(w http.ResponseWriter, r *http.Request) {
...
 	var disposition string
 
 	if action == "inline" {
 		disposition = "inline"
 		/*
-		metadata.ContentType is unable to determine the type of the content, 
-		So add text/plain in this case to fix XSS related issues/
+			metadata.ContentType is unable to determine the type of the content,
+			metadata.ContentType is unable to determine the type of the content,
+			metadata.ContentType is unable to determine the type of the content,
```
Comment (stefanbenten) [inline server/handlers.go:1060]: Those can be removed?

**Review by stefanbenten — COMMENTED** [review]
Didnt test with Storj yet sadly. But i'll try to squeeze it in today.

File: server/handlers.go [hunk 2]
```diff
@@ -54,11 +53,13 @@ import (
 	textTemplate "text/template"
 	"time"
 
+	"github.com/dutchcoders/transfer.sh/server/storage"
+
```
[the remark this answers is not in the document]
Comment (aspacca) [reply server/handlers.go:57]: please, let's have a command for applying this kind of lining :)

File: server/storage/common.go [hunk 3]
```diff
@@ -4,13 +4,95 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"strconv"
 	"time"
+
+	"regexp"
```
[the remark this answers is not in the document]
Comment (aspacca) [reply server/storage/common.go:10]: same as above

**Comment by aspacca** [comment]
> Didnt test with Storj yet sadly. But i'll try to squeeze it in today.

I've tried five minutes ago: upload was fine, but when I try to open the url jsondecoding of the metadata fails with `Error metadata: EOF`

The donwloaded metadata seems empty or something similar: that's even before than applying the range request when downloading the real file

btw: how do I delete the storj account I've created? :)

**Comment by aspacca** [comment]
> Can you check if the file actually exists in the bucket?

they exists, and the metadata is a proper json: if i print the content from storj.Download i get `(0x0, 0x0)` or something similar

File: server/storage/common.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ochaton) [reply server/storage/common.go:10]: `go fmt` is reordering imports by default :( maybe you have some configuration for it to get along with the codestyle?

[the remark this answers is not in the document]
Comment (aspacca) [reply server/storage/common.go:10]: i use directly `gofmt`: I don't think to have any configuration for it

[the remark this answers is not in the document]
Comment (aspacca) [reply server/storage/common.go:10]: is it indeed `goimports -w .` but it does not change anything of the above

**Comment by aspacca** [comment]
> I've tried five minutes ago: upload was fine, but when I try to open the url jsondecoding of the metadata fails with `Error metadata: EOF`

it was because we didn't end in https://github.com/storj/uplink/blob/main/download.go#L60-L62, but rather in https://github.com/storj/uplink/blob/main/download.go#L78-L82, and finally reaching https://github.com/storj/uplink/blob/main/private/stream/download.go#L72

it seems that if you want the full range either you pass `nil` as `options *DownloadOptions` in `Project.DownloadObject()` or you have to calculate the proper `DownloadOptions.Offset` and `DownloadOptions.Length`. not sure how it worked before and if it's related to dump of the storj depdencies

good to merge for me, please have a final look, @stefanbenten

**Comment by ochaton** [comment]
@aspacca maybe we should provide docker-compose.yml or something like that to do integration testing?

**Comment by aspacca** [comment]
@ochaton I've been thinking for a while about having integration tests: the problem is that only local and s3 storage are possible to test with incurring in actual infra costs for the integration :(

**Comment by ochaton** [comment]
@aspacca It seems that after refactoring we accidentally removed setting Content-Length header for Range downloads.

Caught on my installation during download for large range requests (64Mbytes in the middle of the file).

It is enough to add this line:
https://github.com/ochaton/transfer.sh/blob/main/server/handlers.go#L1207

It is missing in upstream now https://github.com/dutchcoders/transfer.sh/blob/main/server/handlers.go#L1206

so now in this line https://github.com/dutchcoders/transfer.sh/blob/main/server/handlers.go#L1236
headers are sent to client without necessary headers.

This works, because, golang server sets `Transfer-Encoding: chunked instead` and this breaks some download managers (I'm using aria2c).

Could you patch this place in the upstream?

**Pull request merged**